cmake/toolchain/oneApi/generic.cmake

```diff
@@ -1,9 +1,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if($ENV{ONEAPI_ROOT})
-  set_ifndef(ONEAPI_TOOLCHAIN_PATH "$ENV{ONEAPI_ROOT}")
+if(DEFINED ENV{ONEAPI_ROOT})
+  set_ifndef(ONEAPI_TOOLCHAIN_PATH $ENV{ONEAPI_ROOT})
```

Comment:
should `ONEAPI_ROOT` also use `zephyr_get`?

Reply:
this is because the `ONEAPI_ROOT` is a standard environment setting even when using One API outside Zephyr, which is why we support it.

But as it doesn't follow the regular Zephyr toolchain settings, then I think we should only allow this as a special case in the environment.

I believe @nashif can provide more info on this.

See also: https://github.com/zephyrproject-rtos/zephyr/pull/31706#discussion_r596664076 (may need to expand resolved comments for link to work in that PR)